main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -945,4 +946,72 @@ object Type {
     }
   }
 
+  /**
+    * Checks that `tpe1` is a subtype of `tpe2`.
+    *
+    * `tpe1` is a subtype of `tpe2` if:
+    *
+    * (a) `tpe1` has the exact same flix type as `tpe2`
+    *
+    * (b) both types are java types and `tpe1` is a subtype of `tpe2`
+    *
+    * (c) both types are functions and `tpe1` is a subtype of `tpe2`
+    *
+    * AND
+    *
+    * the purity of the expression is being cast from `pure` -> `ef` -> `impure`.
+    *
+    * OR
+    *
+    * the effect set of the expression is a subset of the effect set being cast to.
+    *
+    */
+  def isRegularSubType(tpe1: Type, tpe2: Type, renv: RigidityEnv)(implicit flix: Flix): Boolean = (tpe1.baseType, tpe2.baseType) match {
```

Comment:
I would change these to say each Kind explicitly: `isStarSubtype`, `isBoolSubtype`, and `isEffectSubtype`, so it's immediately clear:
1. the functions are doing the same idea
2. what kind it can operate on

The description of each of these (subset/"more pure") can go in the comments/documentation.

Reply:
Ahh great, thanks :)